var/spack/repos/builtin/packages/xgboost/package.py

```diff
@@ -0,0 +1,97 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+from shutil import copyfile
+import glob
+
+
+class Xgboost(Package):
+    """
+    XGBoost is an optimized distributed gradient boosting library designed to
+    be highly efficient, flexible and portable. It implements machine learning
```

Comment:
Can you write the docstring using a short first sentence and then, in a new paragraph, an optional broader description? What I'd like is something similar to:
```python
class Xgboost(Package):
    """XGBoost is an optimized distributed gradient boosting library designed to
    be highly efficient, flexible and portable. 

    It implements machine learning ...
    """
```
The reson is that in many contexts (e.g. modules) the first sentence is used as a short description of the package, while the entire docstring as a long one.

Reply:
How does Spack handle this? Is it stopping once it finds the first period? In that case, does the extra blank line matter?

Reply:
Yes, it matters:
```python
def short_description(self):
    # If we have a valid docstring return the first paragraph.
    docstring = type(self.spec.package).__doc__
    if docstring:
        value = docstring.split('\n\n')[0]
        # Transform tabs and friends into spaces
        value = re.sub(r'\s+', ' ', value)
        # Turn double quotes into single quotes (double quotes are needed
        # to start and end strings)
        value = re.sub(r'"', "'", value)
        return value
    # Otherwise the short description is just the package + version
    return self.spec.format("$_ $@")
```